string-template/src/main/java/org/jdbi/v3/stringtemplate/UseStringTemplateSqlLocator.java

```diff
@@ -58,9 +64,20 @@
     Class<? extends StatementRewriter> value() default ColonPrefixStatementRewriter.class;
 
     class LocatorFactory implements SqlObjectConfigurerFactory {
+        private static final Map<String, Class<?>> TYPE_CACHE = new ConcurrentHashMap<>();
+
         private static final SqlLocator SQL_LOCATOR = (sqlObjectType, method) -> {
             String name = SqlAnnotations.getAnnotationValue(method).orElseGet(method::getName);
-            return StringTemplateSqlLocator.findStringTemplateSql(sqlObjectType, name);
+            StringTemplateGroup group = findStringTemplateGroup(sqlObjectType);
+            if (!group.isDefined(name)) {
+                throw new IllegalStateException("No StringTemplate group " + name + " for class " + sqlObjectType);
+            }
+
+            String typeName = sqlObjectType.getName();
+            TYPE_CACHE.putIfAbsent(typeName, sqlObjectType);
+
+            // Hash character # should never appear in class names.. right?
+            return typeName + "#" + name;
```

Comment:
From what I can tell in the java language spec, we're good. Only UTF letters and digits are valid in class names.